src/format/xVecLibBoomerangFormatV0_1/index.tsx

```diff
@@ -0,0 +1,138 @@
+import { makeAutoObservable } from "mobx";
+import { MainApp, getAppStores } from "@core/MainApp";
+import { makeId } from "@core/Util";
+import { Path, Segment } from "@core/Path";
+import { UnitOfLength, UnitConverter, Quantity } from "@core/Unit";
+import { GeneralConfig, convertFormat } from "../Config";
+import { Format, importPDJDataFromTextFile } from "../Format";
+import { PointCalculationResult, getPathPoints } from "@core/Calculation";
+import { GeneralConfigImpl } from "./GeneralConfig";
+import { PathConfigImpl } from "./PathConfig";
+import { SmartBuffer } from "smart-buffer";
+// observable class
+export class xVecLibBoomerangFormatV0_1 implements Format {
+  isInit: boolean = false;
+  uid: string;
+
+  protected gc = new GeneralConfigImpl(this);
+
+  constructor() {
+    this.uid = makeId(10);
+    makeAutoObservable(this);
+  }
+
+  createNewInstance(): Format {
+    return new xVecLibBoomerangFormatV0_1();
+  }
+
+  getName(): string {
+    return "xVecLib Boomerang v0.1.0 (inch)";
+  }
+  getDescription(): string {
+    return "Generates a sequence of xVecLib .moveToBoom function calls.";
+  }
+  register(app: MainApp): void {
+    if (this.isInit) return;
+    this.isInit = true;
+  }
+
+  unregister(): void {}
+
+  getGeneralConfig(): GeneralConfig {
+    return this.gc;
+  }
+
+  createPath(...segments: Segment[]): Path {
+    return new Path(new PathConfigImpl(this), ...segments);
+  }
+
+  getPathPoints(path: Path): PointCalculationResult {
+    const result = getPathPoints(path, new Quantity(this.gc.pointDensity, this.gc.uol));
+    return result;
+  }
+
+  convertFromFormat(oldFormat: Format, oldPaths: Path[]): Path[] {
+    return convertFormat(this, oldFormat, oldPaths);
+  }
+
+  importPathsFromFile(buffer: ArrayBuffer): Path[] {
+    throw new Error("Unable to import paths from this format, try other formats?");
+  }
+
+  exportCode(): string {
+    const { app } = getAppStores();
+    let rtn = "";
+    let arr = [];
+
+    const gc = app.gc as GeneralConfigImpl;
+
+    const path = app.interestedPath();
+    if (path === undefined) throw new Error("No path to export");
+    if (path.segments.length === 0) throw new Error("No segment to export");
+
+    const uc = new UnitConverter(this.gc.uol, UnitOfLength.Inch);
+    const points = path.segments;
+    if (points.length > 0) {
+      for (const point of points) {
+        if (path.segments.indexOf(point) === 0) {
+          console.log(point.first.heading);
+          let tmpp = point.first.heading > 180 ? point.first.heading - 360 : point.first.heading;
+
+          rtn += `${gc.chassisName}.setPos(${Math.round(point.first.x)}, ${Math.round(point.first.y)},${tmpp});\n`;
+          rtn += `${gc.chassisName}.printCoords();\n`;
+        }
+        let pnt = 0;
+        let last = path.controls.at(path.controls.length - 1);
+        let dis;
+        if (last != null) {
```

Comment:
`path.controls` should always have 2 or 4 controls. This if statement is not necessary.